File: core/crates/primitives/src/testkit/application_metadata_mock.rs
use crate::{ApplicationMetadata, ApplicationMetadataSource};

impl ApplicationMetadata {
    pub fn mock() -> Self {
        ApplicationMetadata {
            name: "Test Dapp".to_string(),
            description: "Test Dapp".to_string(),
            url: "https://example.com".to_string(),
            icon: "https://example.com/icon.png".to_string(),
            source: ApplicationMetadataSource::WalletConnect,
        }
    }
}

File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/ApplicationMetadataMock.kt
package com.gemwallet.android.testkit

import com.wallet.core.primitives.ApplicationMetadata
import com.wallet.core.primitives.ApplicationMetadataSource

fun mockApplicationMetadata(name: String = "Uniswap", source: ApplicationMetadataSource = ApplicationMetadataSource.WalletConnect) = ApplicationMetadata(
    name = name,
    description = "Swap",
    url = "https://app.uniswap.org",
    icon = "https://app.uniswap.org/icon.png",
    source = source,
)
